app/middleware/otp_verify.js

```diff
@@ -36,7 +36,9 @@ function verifyOtpForServiceInvite (req, res, next) {
   }
 
   return validateOtp(otpCode)
-    .then(() => registrationService.submitServiceInviteOtpCode(code, otpCode, correlationId))
+    .then(() => {
+      return registrationService.submitServiceInviteOtpCode(code, otpCode, correlationId)
+    })
```

Comment:
Why is this better? More verbose?

Reply:
In order for it to work against the promise 'catch' statement in that promise chain, I had to investigate why the previous incarnation wouldn't work with native promises, so had to explicitly do it otherwise unit tests that rejected the underlying promise were never caught.